practical1/lib/models/user_bloc.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:async';
+import 'user.dart';
+import 'package:practical1/networking.dart';
+import 'package:bloc/bloc.dart';
+import 'package:meta/meta.dart';
+import 'package:practical1/models/user.dart';
+
+part 'user_event.dart';
+part 'user_state.dart';
+
+class UserBloc extends Bloc<UserEvent, UserState> {
+  UserBloc() : super(UserInitial());
+
+  List<User> allUsers = [];
+  int from = 0;
+
+  @override
+  Stream<UserState> mapEventToState(
+    UserEvent event,
+  ) async* {
+    if (event is GetData) {
+      if (allUsers.length == 0) {
+        yield UserLoading();
+      }
+      Future allData = getData(from);
+      for (var x in await allData) {
+        allUsers.add(User(x['login'], x['avatar_url']));
+      }
+      from += 12;
```

Comment:
Rather static, this should be incremented for the length of the users received from the API